.github/workflows/dash.yaml

```diff
@@ -0,0 +1,28 @@
+name: dash
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  check-dash:
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      # taken from here: https://github.com/eclipse/dash-licenses#example-python
+      - name: Create Dash Dependency Report
+        run: |
+          cat kuksa-client/requirements.txt | grep -v \# \
+          | sed -E -e 's|([^= ]+)==([^= ]+)|pypi/pypi/-/\1/\2|' -e 's| ||g' \
+          | sort | uniq \
+          > dependencies.txt
```

Comment:
I tested out both reading dependencies from requirements.txt as well as reading it from a (newly created) virtual environment using pipdeptree. I decided for reading the dependencies because it sounds more reasonable and because the other one contained "junk" which clearly should not belong there. 

```
-egit+ssh://git@github.com/eclipse-kuksa/kuksa-python-sdk.git@57b7ec81902a03aa6f216536d8537764a7c0aa8b#egg=kuksa_client&subdirectory=kuksa-client
altgraph@file:///System/Volumes/Data/SWE/Apps/DT/BuildRoots/BuildRoot11/ActiveBuildRoot/Library/Caches/com.apple.xbs/Sources/python3/python3-141/altgraph-0.17.2-py2.py3-none-any.whl
future@file:///System/Volumes/Data/SWE/Apps/DT/BuildRoots/BuildRoot11/ActiveBuildRoot/Library/Caches/com.apple.xbs/Sources/python3/python3-141/future-0.18.2-py3-none-any.whl
macholib@file:///System/Volumes/Data/SWE/Apps/DT/BuildRoots/BuildRoot11/ActiveBuildRoot/Library/Caches/com.apple.xbs/Sources/python3/python3-141/macholib-1.15.2-py2.py3-none-any.whl
mock-service@file:///Users/wba2hi/git/kuksa.val.services/mock_service
pypi/pypi/-/Pygments/2.15.1
pypi/pypi/-/asgiref/3.7.2
pypi/pypi/-/attrs/23.1.0
pypi/pypi/-/cmd2/1.5.0
pypi/pypi/-/colorama/0.4.6
pypi/pypi/-/decorator/5.1.1
pypi/pypi/-/grpcio-tools/1.56.2
pypi/pypi/-/grpcio/1.56.2
pypi/pypi/-/jsonpath-ng/1.5.3
pypi/pypi/-/numpy/1.25.2
pypi/pypi/-/pip/23.3.1
pypi/pypi/-/pipdeptree/2.13.1
pypi/pypi/-/ply/3.11
pypi/pypi/-/prettytable/3.9.0
pypi/pypi/-/protobuf/4.23.4
pypi/pypi/-/pyperclip/1.8.2
pypi/pypi/-/scipy/1.11.2
pypi/pypi/-/setuptools/58.0.4
pypi/pypi/-/six/1.16.0
pypi/pypi/-/sqlparse/0.4.4
pypi/pypi/-/types-protobuf/4.24.0.1
pypi/pypi/-/typing_extensions/4.8.0
pypi/pypi/-/wcwidth/0.2.6
pypi/pypi/-/websockets/11.0.3
wheel@file:///System/Volumes/Data/SWE/Apps/DT/BuildRoots/BuildRoot11/ActiveBuildRoot/Library/Caches/com.apple.xbs/Sources/python3/python3-141/wheel-0.37.0-py2.py3-none-any.whl
```
However, since I'm no Python Expert somebody with more knowledge should take a look at it

Reply:
I think just form requirements does not work well, because you would miss the transient dependencies, i.e. you would not find jsonpath-ng/1.5.3 

In terms of "is Eclipse ok with us using the dependencies here", just the requirements.txt stuff is fine, but in terms of "release the docker", we would need to check everything in the docker (I know it is even more....), so for that it is better to collect all installed dependencies)

Reply:
Since you have more knowledge here, can you do a sanity check on the log from pipdeptree I posted and check if this covers "everything we need"? 

I'm confused here because I would expect stuff like the mock-service and so on to not appear in a new virtual environment, but still it is there

Reply:
I do think your environment was not clear, here is, what I did:

Install kuksa-client in a clean envrionment

```
cd testkc
pipenv --rm
pipenv shell # this will also activate the new empty venv
pip install ../kuksa-python-sdk/kuksa-client
```
 also (needed later), lets see which python is used here

```
(testkc) scs2rng@RNG-C-001JT testkc % which python
/Users/scs2rng/.local/share/virtualenvs/testkc-fNFfcNs5/bin/python
```

to avoid poisoning that environment with pipdeptree you can install it globally/outside the venv. I choose a new one

```
cd pdt 
pipenv shell  
pip install pipdeptree
```

then run pipdeptree with the python environment created earlier


```
(pdt) scs2rng@RNG-C-001JT pdt % pipdeptree --python /Users/scs2rng/.local/share/virtualenvs/testkc-fNFfcNs5/bin/python

kuksa-client==0.4.2.post3+git.b65f5168
├── cmd2 [required: >=1.4,<2.0, installed: 1.5.0]
│   ├── attrs [required: >=16.3.0, installed: 23.1.0]
│   ├── colorama [required: >=0.3.7, installed: 0.4.6]
│   ├── pyperclip [required: >=1.6, installed: 1.8.2]
│   └── wcwidth [required: >=0.1.7, installed: 0.2.12]
├── grpcio-tools [required: >=1.54.2, installed: 1.60.0]
│   ├── grpcio [required: >=1.60.0, installed: 1.60.0]
│   ├── protobuf [required: >=4.21.6,<5.0dev, installed: 4.25.1]
│   └── setuptools [required: Any, installed: 68.2.2]
├── jsonpath-ng [required: >=1.5.3, installed: 1.6.0]
│   └── ply [required: Any, installed: 3.11]
├── pygments [required: >=2.15, installed: 2.17.2]
└── websockets [required: >=10.1, installed: 12.0]
pip==23.3.1
wheel==0.41.3
(pdt) scs2rng@RNG-C-001JT pdt % 
```

Sanity check with pip-licenses

```
(pdt) scs2rng@RNG-C-001JT pdt % pip-licenses --python /Users/scs2rng/.local/share/virtualenvs/testkc-fNFfcNs5/bin/python
 Name          Version                   License                 
 Pygments      2.17.2                    BSD License             
 attrs         23.1.0                    MIT License             
 cmd2          1.5.0                     MIT License             
 colorama      0.4.6                     BSD License             
 grpcio        1.60.0                    Apache Software License 
 grpcio-tools  1.60.0                    Apache Software License 
 jsonpath-ng   1.6.0                     Apache Software License 
 kuksa-client  0.4.2.post3+git.b65f5168  Apache Software License 
 ply           3.11                      BSD                     
 protobuf      4.25.1                    3-Clause BSD License    
 pyperclip     1.8.2                     BSD License             
 websockets    12.0                      BSD License             
(pdt) scs2rng@RNG-C-001JT pdt % 
```

With Python it is never an "exact science" as the specifc versions installed may change depending on python version, but firstly we later could decide to even to to for specific version, or - as this list might be relevant when shipping artifacts -  run it in the container.

Reply:
Thanks for the feedback. I can now reproduce it. Issue was that I had some system-wide aliases for pip and python which basically always lead to using the "global system" environment instead of the "newly created one" when installing pip packages